### ISSUE TITLE
OpenSSHClientTestCase should use procutils.which to locate the ssh client, and should skip sooner

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3095]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)| @therve reported|
|-|-|
|Trac ID|trac#3095|
|Type|defect|
|Created|2008-03-18 15:28:36Z|
|Branch|https://github.com/twisted/twisted/tree/ssh-test-3095|

The current behavior is broken, and it has some weird consequences under windows.

<details><summary>Searchable metadata</summary>

```
trac-id__3095 3095
type__defect defect
reporter__therve therve
priority__normal normal
milestone__ 
branch__branches_ssh_test_3095 branches/ssh-test-3095
branch_author__ 
status__closed closed
resolution__fixed fixed
component__conch conch
keywords__ 
time__1205854116000000 1205854116000000
changetime__1205939605000000 1205939605000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_1"></a>|@therve commented|
|-|-|

(In [22953]) Branching to 'ssh-test-3095'

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_2"></a>|@therve removed owner|
|-|-|

Please review!

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun set owner to @therve|
|-|-|

Looks good, please merge

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_4"></a>|@therve set status to `closed`|
|-|-|

(In [22976]) Merge ssh-test-3095

Author: therve
Reviewer: exarkun
Fixes [#3095](https://github.com/twisted/twisted/issues/3095)

Make OpenSSHClientTestCase use procutils.which instead of manual guessing for
finding the ssh command-line client.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
